python/tank/log.py

```diff
@@ -100,69 +212,96 @@ def get_root_logger(cls):
+        When you start an engine via the :meth:`sgtk.platform.start_engine` method,
+        a file handler will automatically be created if one doesn't already exist.
 
         If you are manually launching toolkit, we recommend that you call
-        this method to initialize logging to file. Calling it multiple times
-        will not result in the information being written to multiple different
-        files - only one file logger can exist per Toolkit session.
+        this method to initialize logging to file early on in your setup.
+        Calling it multiple times will not result in the information being
+        written to multiple different files - only one file logger can
+        exist per session.
 
         :param log_name: Name of logger to create. This will form the
-                         filename of the log file.
+                         filename of the log file. If you pass None, you
+                         ensure that any existing base file handlers terminate
+                         their logging.
+        :returns: The name of the previous log_name that is being switched away from.
         """
-        # avoid cyclic references
-        from .util import LocalFileStorageManager
-        from .util import filesystem
 
-        if cls._std_file_handler:
+        previous_log_name = self._std_file_handler_log_name
+        log.debug("Switching file based std logger from %s to %s" % (previous_log_name, log_name))
+
+        if self._std_file_handler:
             # there is already a log handler.
             # terminate previous one
-            sgtk_root_logger.removeHandler(cls._std_file_handler)
-            cls._std_file_handler = None
+            log.debug(
+                "Tearing down existing log handler '%s' (%s)" % (previous_log_name, self._std_file_handler)
+            )
+            self._root_logger.removeHandler(self._std_file_handler)
+            self._std_file_handler = None
 
-        # set up logging root folder
-        log_folder = LocalFileStorageManager.get_global_root(LocalFileStorageManager.LOGGING)
-        filesystem.ensure_folder_exists(log_folder)
+        # store new log name
+        self._std_file_handler_log_name = log_name
 
-        # generate log path
-        log_file = os.path.join(
-            log_folder,
-            "%s.log" % filesystem.create_valid_filename(log_name)
-        )
+        if log_name:
+            # set up a new handler
 
-        cls._std_file_handler = logging.handlers.RotatingFileHandler(
-            log_file,
-            maxBytes=4096*1024,
-            backupCount=5
-        )
+            # avoid cyclic references
+            from .util import filesystem
+
+            # set up logging root folder
+            filesystem.ensure_folder_exists(self.log_folder)
+
+            # generate log path
+            log_file = os.path.join(
+                self.log_folder,
+                "%s.log" % filesystem.create_valid_filename(log_name)
+            )
```

Comment:
Do we know enough at this point to be able to pass in an override for the log file name?

I'm thinking through the various launch paths and how it would be annoying to have to debug initialization issues and correlate logging between the target engine and one of bootstrap, tk-shell, tk-shotgun, and tk-desktop.

I'm not sure if it is possible, but it would be nice if the entirety of the startup logic could be in the log file for the engine we're going to be running if defined.  So for something like the rv plugin everything could just go into tk-rv.log (like desktop does now).

Thoughts?  Possible?